modules/schema/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql

```diff
@@ -3614,6 +3614,27 @@ input StepConfigSmartGcalInput {
   smartGcalType: SmartGcalType!
 }
 
+"""
+Execution state of a recorded step.
+"""
+enum StepExecutionState {
+
+  "No events have been received since the step was recorded."
+  NOT_STARTED
+
+  "Events have been received, but no terminal `Abort`, `EndStep` or `Stop` events."
+  ONGOING
+
+  "An event with an 'ABORT' 'StepStage' was received."
+  ABORTED
+
+  "An event with an 'END_STEP' 'StepStage' was received."
+  COMPLETED
+
+  "An event with a 'STOP' 'StepStage' was received."
+  STOPPED
```

Comment:
An `ABANDONED` state might make sense as well, but it would require something other than just step events to determine.

Reply:
Would we need to do a sweep after the fact and mark steps as abandoned if we never got a final event?

Reply:
Yeah I think that's right.  A daily sweep of steps that are still `ONGOING`.  I'll leave it for now unless @rpiaggio thinks it is needed?

Reply:
I plan to just hide the `ONGOING` step in observe, and the atom where an `ONGOING` step is located in explore.

So, without a daily sweep we would be hiding executed steps from abandoned atoms. Otherwise, we can check the starting instant and assume that an `ONGOING` atom from the current night is executing, and if it's from a previous night then it is abandoned.